lib/spack/spack/build_systems/autotools.py

```diff
@@ -803,14 +816,23 @@ def remove_libtool_archives(self):
             return
 
         # Remove the files and create a log of what was removed
-        libtool_files = fs.find(str(self.prefix), "*.la", recursive=True)
+        libtool_files = fs.find(str(self.pkg.prefix), "*.la", recursive=True)
         with fs.safe_remove(*libtool_files):
             fs.mkdirp(os.path.dirname(self._removed_la_files_log))
             with open(self._removed_la_files_log, mode="w") as f:
                 f.write("\n".join(libtool_files))
 
+    def setup_build_environment(self, env):
+        if self.spec.platform == "darwin" and macos_version() >= Version("11"):
+            # Many configure files rely on matching '10.*' for macOS version
+            # detection and fail to add flags if it shows as version 11.
+            env.set("MACOSX_DEPLOYMENT_TARGET", "10.16")
+
     # On macOS, force rpaths for shared library IDs and remove duplicate rpaths
-    run_after("install")(PackageBase.apply_macos_rpath_fixups)
+    spack.builder.run_after("install", when="platform=darwin")(apply_macos_rpath_fixups)
+
+
+# FIXME (MULTIPLE BUILD SYSTEMS): Convert packages defining "with_or_without_..."?
```

Comment:
Is this still a FIXME or is this vestigial?

Reply:
Removed the FIXME, everything works as expected.